cpp/src/spatial/point_in_polygon.cu

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/utilities/cuda.cuh>
+#include <cudf/column/column_view.hpp>
+
+#include <cudf/utilities/legacy/type_dispatcher.hpp>
+#include <utilities/legacy/cuda_utils.hpp>
+#include <type_traits>
+#include <utility/legacy/utility.hpp>
+#include <cuspatial/legacy/point_in_polygon.hpp>
+#include <cuspatial/error.hpp>
+
+#include <cudf/legacy/column.hpp>
+#include "cudf/utilities/type_dispatcher.hpp"
+#include "rmm/mr/device/device_memory_resource.hpp"
+
+namespace {
+
+template <typename T>
+__global__ void point_in_polygon_kernel(cudf::size_type num_test_points,
+                                        const T* const __restrict__ test_points_x,
+                                        const T* const __restrict__ test_points_y,
+                                        cudf::size_type num_polys,
+                                        const cudf::size_type* const __restrict__ poly_offsets,
+                                        const cudf::size_type* const __restrict__ poly_ring_offsets,
+                                        const T* const __restrict__ poly_points_x,
+                                        const T* const __restrict__ poly_points_y,
+                                        cudf::size_type* const __restrict__ result)
+{
+    cudf::size_type idx = blockIdx.x * blockDim.x + threadIdx.x;
+
+    if (idx > num_test_points) {
+        return;
+    }
+
+    cudf::size_type hit_mask = 0;
```

Comment:
cudf::size_type should be used to store things that are limited by the size of a cuDF column. It shouldn't be used just for an integer that needs to be limited to 32 bits. It could change in the future, is this function robust to that?

Note that cuDF doesn't use size_type for bitmasks, which this is. Suggest using an unsigned type.

Reply:
I started with `uint32_t`, but found out `cudf::column` doesn't support unsigned types... ? `cudf::bitmask_type` is `uint32_t`, but that's stored as a `device_buffer` and can be materialized to a `bool` column.

Do we support unsigned columns, and I'm just missing something?